examples/adversarial_training_FBF.py

```diff
@@ -0,0 +1,188 @@
+            layers.append(block(self.in_planes, planes, stride))
+            self.in_planes = planes * block.expansion
+        return nn.Sequential(*layers)
+
+    def forward(self, x):
+        out = self.conv1(x)
+        out = self.layer1(out)
+        out = self.layer2(out)
+        out = self.layer3(out)
+        out = self.layer4(out)
+        out = F.avg_pool2d(out, 4)
+        out = out.view(out.size(0), -1)
+        out = self.linear(out)
+        return out
+
+
+def PreActResNet18():
+    return PreActResNet(PreActBlock, [2, 2, 2, 2])
+
+
+def initialize_weights(module):
+    if isinstance(module, nn.Conv2d):
+        n = module.kernel_size[0] * module.kernel_size[1] * module.out_channels
+        module.weight.data.normal_(0, math.sqrt(2. / n))
+        if module.bias is not None:
+            module.bias.data.zero_()
+    elif isinstance(module, nn.BatchNorm2d):
+        module.weight.data.fill_(1)
+        module.bias.data.zero_()
+    elif isinstance(module, nn.Linear):
+        module.bias.data.zero_()
+
+
+# Step 1: Load the CIFAR10 dataset
+(x_train, y_train), (x_test, y_test), min_pixel_value, max_pixel_value = load_cifar10()
+
+# prepare the tensors for preprocessing
+cifar_mu = np.ones((3, 32, 32))
+cifar_mu[0, :, :] = 0.4914
+cifar_mu[1, :, :] = 0.4822
+cifar_mu[2, :, :] = 0.4465
+
+cifar_std = np.ones((3, 32, 32))
+cifar_std[0, :, :] = 0.2471
+cifar_std[1, :, :] = 0.2435
+cifar_std[2, :, :] = 0.2616
+
+x_train = x_train.transpose(0, 3, 1, 2).astype('float32')
+x_test = x_test.transpose(0, 3, 1, 2).astype('float32')
+
+# Step 2: create the PyTorch model
+model = PreActResNet18()
+# For running on GPU replace the model with the
+# model = PreActResNet18().cuda()
+
+model.apply(initialize_weights)
+model.train()
+
+opt = torch.optim.SGD(model.parameters(), lr=0.21, momentum=0.9, weight_decay=5e-4)
+
+# if you have apex installed, the following line should be uncommented for faster processing
+# model, opt = amp.initialize(model, opt, opt_level="O2", loss_scale=1.0, master_weights=False)
+
+criterion = nn.CrossEntropyLoss()
+# Step 3: Create the ART classifier
+
+classifier = PyTorchClassifier(
+    model=model,
+    clip_values=(0.0, 1.0),
+    preprocessing=(cifar_mu, cifar_std),
+    loss=criterion,
+    optimizer=opt,
+    input_shape=(3, 32, 32),
+    nb_classes=10,
+)
+
+# Step 4: Create the trainer object - AdversarialTrainerFBFPyTorch
+# if you have apex installed, change use_amp to True
+epsilon = (8.0 / 255.)
+trainer = AdversarialTrainerFBFPyTorch(classifier, eps=epsilon, use_amp=False)
+
+# Build a Keras image augmentation object and wrap it in ART
+batch_size = 128
+datagen = ImageDataGenerator(
+    horizontal_flip=True
+)
+
+datagen.fit(x_train)
+art_datagen = KerasDataGenerator(
+    datagen.flow(x=x_train, y=y_train, batch_size=batch_size, shuffle=True),
+    size=x_train.shape[0],
+    batch_size=batch_size,
+)
+
+# Step 5: fit the trainer
+trainer.fit_generator(art_datagen, nb_epochs=30)
```

Comment:
Would it be possible to extend this example here with and evasion attack to show the adversarial accuracy of the original and robust models?